### ISSUE TITLE
isotropic_remeshing segfaults even when is_valid_polygon_mesh is true

### ISSUE THREAD

**Issue by jasjuang** [CGAL/cgal#4112]
##  Issue Details

I ran the function `CGAL::Polygon_mesh_processing::isotropic_remeshing` for a particular mesh that I have and it causes a segfault, and I am not sure why. What's even more confusing is when I check it with `CGAL::is_valid_polygon_mesh` it appears to be true. Please see below minimal code to reproduce the problem. 

```
typedef CGAL::Simple_cartesian<double> K;
typedef CGAL::Surface_mesh<K::Point_3> Mesh;

Mesh m;

std::ifstream input("test.off");
input >> m;
input.close();

std::cout << CGAL::is_valid_polygon_mesh(m) << std::endl;

CGAL::Polygon_mesh_processing::isotropic_remeshing(faces(m), 5.0, m);
```

Here is the mesh that triggers the failure. [test.zip](https://github.com/CGAL/cgal/files/3432733/test.zip) Additionally, this raw mesh is reconstructed from multiple depth sensors, so when I run `isotropic_remeshing` on the previous frame where the raw mesh is nearly identical as this one (the only difference is a slight head movement), it actually doesn't segfault, which makes it even more confusing. Please let me know if you can reproduce the problem.

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Ubuntu 18.04
* Compiler: GCC 7
* Release or debug mode: Release 
* CGAL version: Latest master
* Boost version: 1.68

**Comment by janetournois** [comment]
Thanks @jasjuang for your report. I can reproduce the seg fault. I will take a look at try to fix it.

**Comment by janetournois** [comment]
This bug should be fixed by PR #4129.
Thanks again for the report!

**Comment by jasjuang** [comment]
@janetournois I can confirm that PR #4129 fixes the problem. Thanks for the fix and please merge!
